src/toxcore/dht/server/mod.rs

```diff
@@ -1461,6 +1492,12 @@ mod tests {
     const ONION_RETURN_2_PAYLOAD_SIZE: usize = ONION_RETURN_2_SIZE - secretbox::NONCEBYTES;
     const ONION_RETURN_3_PAYLOAD_SIZE: usize = ONION_RETURN_3_SIZE - secretbox::NONCEBYTES;
 
+    impl Server {
+        pub fn has_friend(&self, pk: &PublicKey) -> bool {
+            self.friends.read().contains_key(pk)
```

Comment:
I doubt whether dht::Server should not answer whether it has friend or not.

Reply:
What do you mean?

Reply:
I think it's a responsibility of Messenger, not dht::Server.

Reply:
But it's for test purposes only - it's located inside `tests` module. I use it to check if friends module really adds/removes friends into/from internal modules.

Reply:
Oh sorry